### PR TITLE
[ROCm][CI] add Aiter ops tests

### PR DESCRIPTION
**Pull request by divakar-amd**
Add ROCm AITER ops tests and enable `Kernels Core Operation Test` for the MI355 mirror

 Tests cover:
  - rms_norm and rms_norm2d_with_add vs PyTorch reference
  - triton_rotary_embedding vs manual NeoX RoPE reference (xfail - known precision gap)
  - act_mul_and_fp8_group_quant (SiGLU + FP8 group quant)
  - Fused RMSNorm + quantization ops vs sequential composition
  - End-to-end inference chain (RMSNorm → FP8 quant → dequant)

 New ops added to rocm_aiter_ops:
  - rms_norm - wraps aiter.rmsnorm2d_fwd
  - rms_norm2d_with_add - wraps aiter.rmsnorm2d_fwd_with_add

### CODE REVIEW

File: tests/kernels/core/test_rocm_aiter_ops.py [hunk 1]
```diff
@@ -0,0 +1,627 @@
...
+    torch.set_default_device("cuda")
+    torch.manual_seed(0)
+
+
+def require_aiter():
+    from vllm._aiter_ops import is_aiter_found_and_supported
+
+    assert is_aiter_found_and_supported(), (
+        "aiter is required on supported ROCm hardware for this test"
+    )
```
Comment (AndreasKaratzas) [inline tests/kernels/core/test_rocm_aiter_ops.py]: Probably best if we just skip if aiter is not found and supported (eg mi250 arch)

File: tests/kernels/core/test_rocm_aiter_ops.py [hunk 2]
```diff
@@ -0,0 +1,627 @@
...
+pytestmark = pytest.mark.skipif(
+    not current_platform.is_rocm(), reason="ROCm-specific tests"
+)
+
+
+@pytest.fixture(autouse=True)
+def setup_cuda_device():
+    """Set default device to CUDA and seed RNG for all tests."""
+    torch.set_default_device("cuda")
+    torch.manual_seed(0)
```
Comment (AndreasKaratzas) [inline tests/kernels/core/test_rocm_aiter_ops.py]: probably best if we use the seed utility from tests/utils

File: tests/kernels/core/test_rocm_aiter_ops.py [hunk 3]
```diff
@@ -0,0 +1,627 @@
...
+    # Reference: float32 RMSNorm for precision
+    ref = _rms_norm_reference(x, weight, eps).to(dtype)
+
+    out = rocm_aiter_ops.rms_norm(x, weight, eps)
+    _assert_close_budget(
+        out.float(),
+        ref.float(),
+        label=f"rms_norm dtype={dtype}",
+        atol=1e-2,
+        rtol=1e-2,
```
Comment (AndreasKaratzas) [inline tests/kernels/core/test_rocm_aiter_ops.py]: are the tolerances checked? are they numerically also correct? i m sure the test passes but i want to make sure that these tolerances also numerically make sense (aka how much such tolerance for fp16 should theoretically be based on ULP logic)?

File: tests/kernels/core/test_rocm_aiter_ops.py [hunk 4]
```diff
@@ -0,0 +1,627 @@
...
+    out, res_out = rocm_aiter_ops.rms_norm2d_with_add(x, residual, weight, eps)
+
+    _assert_close_budget(
+        out.float(),
+        ref_normed.float(),
+        label="rmsnorm_with_add normed",
+        atol=1e-2,
+        rtol=1e-2,
+    )
+    torch.testing.assert_close(res_out, ref_residual, atol=1e-2, rtol=1.6e-2)
```
Comment (AndreasKaratzas) [inline tests/kernels/core/test_rocm_aiter_ops.py]: same

File: tests/kernels/core/test_rocm_aiter_ops.py [hunk 5]
```diff
@@ -0,0 +1,627 @@
...
+
+
+@pytest.mark.xfail(
+    strict=True,
+    raises=AssertionError,
+    reason=(
+        "AITER Triton RoPE precision gap: max element error ~0.03125 (2^-5 BF16 ULP) "
+        "requires atol~=2e-2; NVIDIA CUDA RoPE achieves atol=1e-3 for bf16 "
+        "(allclose_default.py). Fix in upstream aiter rope kernel."
```
Comment (AndreasKaratzas) [inline tests/kernels/core/test_rocm_aiter_ops.py]: can you open a ticket on AITER and have it linked here in this very string?

**Review by AndreasKaratzas — COMMENTED** [review]
Overall LGTM, but let's address these few comments carefully.

File: tests/kernels/core/test_rocm_aiter_ops.py (hunk 3 above)
Comment (divakar-amd) [reply tests/kernels/core/test_rocm_aiter_ops.py]: Agreed. Updated to ULP based comparison and the default torch values.

File: tests/kernels/core/test_rocm_aiter_ops.py (hunk 5 above)
Comment (divakar-amd) [reply tests/kernels/core/test_rocm_aiter_ops.py]: Done. Here's the aiter issue link: https://github.com/ROCm/aiter/issues/4765

**Review by AndreasKaratzas — APPROVED** [review]
LGTM

File: tests/kernels/core/test_rocm_aiter_ops.py [hunk 6]
```diff
@@ -0,0 +1,627 @@
...
+    cos_cache = torch.cat([cos_half, cos_half], dim=-1)  # [max_pos, head_size]
+    sin_cache = torch.cat([sin_half, sin_half], dim=-1)  # [max_pos, head_size]
+    cos_sin_cache = torch.cat([cos_cache, sin_cache], dim=-1)  # [max_pos, 2*head_size]
+
+    # Reference: NeoX-style pairwise rotation with front-half frequencies only.
+    # rotate_style=0 (NeoX): [x1*c - x2*s, x2*c + x1*s]
+    cos_pos = cos_half[positions]  # [num_tokens, half_dim]
+    sin_pos = sin_half[positions]  # [num_tokens, half_dim]
+
+    def apply_rope_ref(t: torch.Tensor) -> torch.Tensor:
```
Comment (Rohan138) [inline tests/kernels/core/test_rocm_aiter_ops.py:302]: `t.float()` upcasts the operands to float32 before doing the unit RoPE op, which does not match aiter triton (keeps native dtype): https://github.com/ROCm/aiter/blob/main/aiter/ops/triton/_triton_kernels/rope/rope.py#L858

This has been a point of divergence before, because even vLLM's C++ and Triton RoPE impls do not fully agree on upcasting behavior for RoPE.
vLLM C++ always upcasts to fp32: https://github.com/vllm-project/vllm/blob/main/csrc/libtorch_stable/pos_encoding_kernels.cu#L18
vLLM Triton keeps native dtype by default unless `enable_fp32_compute` is set: https://github.com/vllm-project/vllm/blob/main/vllm/model_executor/layers/rotary_embedding/common.py#L151

Comment (divakar-amd) [reply tests/kernels/core/test_rocm_aiter_ops.py:302]: Thanks for sharing this @Rohan138. Opened a new PR to handle the fix.
